File: lecture0-30/lecture010/power.cpp
#include<iostream>

using namespace std;
int power(int a , int b){
    if(b==0)
    return 1;

    int recans=power(a,b-1);
    return (recans)*a;
} 
int main(int args , char** argv){
    cout<<power(2,5);
    
}

File: lecture010/power.java
import java.util.Scanner;

public class power {
    public static Scanner scn = new Scanner(System.in);

    public static void main(String[] args) {
        System.out.println(power(2, 5));
    }

    public static int power(int a, int b) {
        if (b == 0)
            return 1;

        int recAns = power(a, b - 1);
        return recAns * a;
    }
}